FirebaseAdmin/FirebaseAdmin/AppCheck/AppCheckApiClient.cs

```diff
@@ -0,0 +1,198 @@
+        /// Get Verify Token Url .
+        /// </summary>
+        /// <returns>A formatted verify token url.</returns>
+        private string GetVerifyTokenUrl()
+        {
+            var urlParams = new Dictionary<string, string>
+            {
+                { "projectId", this.projectId },
+            };
+
+            string baseUrl = this.FormatString(OneTimeUseTokenVerificationUrlFormat, urlParams);
+            return this.FormatString(baseUrl, null);
+        }
+
+        /// <summary>
+        /// Get url from FirebaseApp Id .
+        /// </summary>
+        /// <param name="appId">The FirebaseApp Id.</param>
+        /// <returns>A formatted verify token url.</returns>
+        private string GetUrl(string appId)
+        {
+            if (string.IsNullOrEmpty(this.projectId))
+            {
+                this.projectId = this.app.GetProjectId();
+            }
+
+            if (string.IsNullOrEmpty(this.projectId))
+            {
+                string errorMessage = "Failed to determine project ID. Initialize the SDK with service account " +
+                      "credentials or set project ID as an app option. Alternatively, set the " +
+                      "GOOGLE_CLOUD_PROJECT environment variable.";
+                throw new ArgumentException(
+                    "unknown-error",
+                    errorMessage);
+            }
+
+            var urlParams = new Dictionary<string, string>
+            {
+                { "projectId", this.projectId },
+                { "appId", appId },
+            };
+            string baseUrl = this.FormatString(ApiUrlFormat, urlParams);
+            return baseUrl;
+        }
+
+        /// <summary>
+        /// Converts a duration string with the suffix `s` to milliseconds.
+        /// </summary>
+        /// <param name="duration">The duration as a string with the suffix "s" preceded by the number of seconds.</param>
+        /// <returns> The duration in milliseconds.</returns>
+        private int StringToMilliseconds(string duration)
+        {
+            if (string.IsNullOrEmpty(duration) || !duration.EndsWith("s"))
+            {
+                throw new ArgumentException("invalid-argument", "`ttl` must be a valid duration string with the suffix `s`.");
+            }
+
+            string modifiedString = duration.Remove(duration.Length - 1);
+            return int.Parse(modifiedString) * 1000;
+        }
+
+        /// <summary>
+        /// Formats a string of form 'project/{projectId}/{api}' and replaces with corresponding arguments {projectId: '1234', api: 'resource'}.
+        /// </summary>
+        /// <param name="str">The original string where the param need to be replaced.</param>
+        /// <param name="urlParams">The optional parameters to replace in thestring.</param>
+        /// <returns> The resulting formatted string. </returns>
+        private string FormatString(string str, Dictionary<string, string> urlParams)
```

Comment:
This is such a generic method that has nothing to do with the app check api client. Please move this to a seperate util class.

Reply:
I have updated the AppCheckApiClient.cs file, and the issue has been resolved